docs/rules/no-hooks-from-ancestor-modules.md

```diff
@@ -0,0 +1,57 @@
+# Forbid the use of hooks from ancestor modules (no-hooks-from-ancestor-modules)
+
+When a QUnit `module` is used with a nested callback, the callback provides a `hooks`
+object as its first argument. This allows calling `hooks.beforeEach` and `hooks.afterEach`
+within that callback's body.
+
+More deeply nested `module` uses should use their own callback's provided `hooks` argument
+and not one from an ancestor `module`.
+
+## Rule Details
+
+The following patterns are considered warnings:
+
+```js
+
+QUnit.module("outer module", function(hooks) {
+  hooks.beforeEach(function() {});
```

Comment:
This line should not warn, so it might be good to simplify this example by removing this line (or maybe just adding a comment saying no warning would appear on this line). That way, the inner `hooks.beforeEach` is the only code that a reader would interpret as having a lint error.